pkg/controller/registry/resolver/resolver_test.go

```diff
@@ -37,56 +37,202 @@ func TestSingleSourceResolveInstallPlan(t *testing.T) {
 	}
 
-	resolver := &SingleSourceResolver{}
-
 	for _, tt := range table {
 		t.Run(tt.description, func(t *testing.T) {
+			namespace := "default"
+
 			log.SetLevel(log.DebugLevel)
 			// Create a plan that is attempting to install the planCSVName.
-			plan := installPlan(tt.planCSVName)
+			plan := installPlan(namespace, tt.planCSVName)
 
 			// Create a catalog source containing a CSVs and CRDs with the provided
 			// names.
 			src := registry.NewInMem()
 			for _, name := range tt.crdNames {
-				err := src.SetCRDDefinition(crd(name))
+				err := src.SetCRDDefinition(crd(name, namespace))
 				require.NoError(t, err)
 			}
 			for _, names := range tt.csv {
 				// We add unsafe so that we can test invalid states
-				src.AddOrReplaceService(csv(names.name, names.owned, names.required))
+				src.AddOrReplaceService(csv(names.name, namespace, names.owned, names.required))
 			}
 
-			srcKey := registry.SourceKey{"tectonic-ocs", plan.Namespace}
+			srcKey := registry.SourceKey{
+				Name:      "tectonic-ocs",
+				Namespace: plan.Namespace,
+			}
 
 			srcMap := map[registry.SourceKey]registry.Source{
 				srcKey: src,
 			}
 
-			// Resolve the plan.
+			// Resolve the plan
 			steps, err := resolver.ResolveInstallPlan(srcMap, srcKey, "alm-catalog", &plan)
 			plan.Status.Plan = steps
 
-			// Assert the error is as expected.
+			// Assert the error is as expected
 			if tt.expectedErr == nil {
 				require.Nil(t, err)
 			} else {
 				require.Equal(t, tt.expectedErr, err)
 			}
 
-			// Assert the number of items in the plan are equal.
+			// Assert the number of items in the plan are equal
 			require.Equal(t, tt.expectedPlanLen, len(plan.Status.Plan))
 
-			// Assert that all StepResources have the have the correct CatalogSource set
+			// Assert that all StepResources have the have the correct catalog source name and namespace set
 			for _, step := range plan.Status.Plan {
 				require.Equal(t, step.Resource.CatalogSource, "tectonic-ocs")
+				require.Equal(t, step.Resource.CatalogSourceNamespace, plan.Namespace)
+			}
+		})
+	}
+}
+
+func multiSourceResolveInstallPlan(t *testing.T, resolver DependencyResolver) {
+
+	// Define some source keys representing different catalog sources (all in same namespace for now)
+	sourceA := registry.SourceKey{Namespace: "default", Name: "tectonic-ocs-a"}
+	sourceB := registry.SourceKey{Namespace: "default", Name: "tectonic-ocs-b"}
+	sourceC := registry.SourceKey{Namespace: "default", Name: "tectonic-ocs-c"}
+
+	type csvName struct {
+		name     string
+		owned    []string
+		required []string
+		srcKey   registry.SourceKey
+	}
+	type crdName struct {
+		name   string
+		srcKey registry.SourceKey
+	}
+	var table = []struct {
+		description string
+		planCSVName string
+		csvs        map[string]csvName
+		crds        map[string]crdName
+		srcKeys     []registry.SourceKey
+		expectedErr error
```

Comment:
Instead of building the expected steps from the input csvs/crds, let's just have a list of expected steps (can include secrets then, too)?

Reply:
In order to test ResolveInstallPlan(...), those fields are being used to populate one or more sources with CSVs and CRDs to be resolved.

Reply:
So then I would build those csvs/crds from the expected steps, store them in their respective sources, and ResolveInstallPlan from there?